disease/query.py

```diff
@@ -301,58 +301,65 @@ def _add_merged_meta(self, response: Dict) -> Dict:
         :return: completed response object.
         """
         sources_meta = {}
-        vod = response["disease_descriptor"]
-        ids = [vod["disease"]] + vod.get("xrefs", [])
-        for concept_id in ids:
-            prefix = concept_id.split(":")[0]
-            src_name = PREFIX_LOOKUP[prefix.lower()]
-            if src_name not in sources_meta:
-                sources_meta[src_name] = self.db.get_source_metadata(src_name)
+        disease = response["disease"]
+        sources = [disease.id.split(":")[0]]
+        if disease.mappings:
+            sources += [m.coding.system for m in disease.mappings]
+
+        for src in sources:
+            try:
+                src_name = PREFIX_LOOKUP[src]
+            except KeyError:
+                # not an imported source
+                continue
+            else:
+                if src_name not in sources_meta:
+                    sources_meta[src_name] = self.db.get_source_metadata(src_name)
         response["source_meta_"] = sources_meta
         return response
 
-    def _add_vod(
-        self, response: Dict, record: Dict, query: str, match_type: MatchType
+    def _add_disease(
+        self, response: Dict, record: Dict, match_type: MatchType
     ) -> NormalizationService:
-        """Format received DB record as VOD and update response object.
+        """Format received DB record as core Disease object and update response object.
 
-        :param Dict response: in-progress response object
-        :param Dict record: record as stored in DB
-        :param str query: query string from user request
-        :param MatchType match_type: type of match achieved
+        :param response: in-progress response object
+        :param record: record as stored in DB
+        :param match_type: type of match achieved
         :return: completed normalized response object ready to return to user
         """
-        vod = {
-            "id": f"normalize.disease:{quote(query)}",
-            "type": "DiseaseDescriptor",
-            "disease": record["concept_id"],
-            "label": record["label"],
-            "extensions": [],
-        }
-        if "xrefs" in record:
-            vod["xrefs"] = record["xrefs"]
+        disease_obj = core_models.Disease(
+            id=record["concept_id"], label=record["label"]
```

Comment:
I'm not sure about the id. Should we add back the prefix with (`normalize.disease`)?

Reply:
I do think the prefix (`normalize.disease`) should be added back. My next questions are:
- What's the suffix? I don't think it should be original query. I think it should either be the concept ID or label
- To indicate the `normalized_concept_id` (? use this name or a diff name?), would we just stick this as an extension?

Reply:
I would really prefer not to put the normalized ID into extensions -- it's probably the most important part of the response. Maybe it could go in a wrapper object on top of the core disease entity?

Reply:
Ya that was my hesitation too and why I kept going back and forth last night. 

How's this?

```
class NormalizationService(BaseModel):
    """Response containing one or more merged records and source data."""

    query: StrictStr
    warnings: Optional[Dict] = None
    match_type: MatchType
    normalized_concept_id: Optional[str] = None  # add this
    disease: Optional[core_models.Disease] = None
    source_meta_: Optional[Dict[SourceName, SourceMeta]] = None
    service_meta_: ServiceMeta
```

and change the `disease` to 

```python
id=f"normalize.disease.{record['concept_id'}"
```

Use `.` rather than `:` to preserve CURIE

Reply:
@jsstevenson not sure about the field name. `normalized_id`? `merged_id`? Something else?

Reply:
Yeah this looks good to me. `normalized_id` fine imo

Reply:
@jsstevenson Thank you! I'll make these changes across gene/disease/therapy